react/providers/I18n/helpers.js

```diff
@@ -0,0 +1,33 @@
+import { initTranslation } from './translation'
+import { initFormat } from './format'
+import { DEFAULT_LANG } from './'
+
+const getAppOrUiLang = defaultLang => {
+  const appContainer = document.querySelector('[role=application]')
+  const cozyUiContainer = {
+    dataset: {
+      cozy: JSON.stringify({
+        locale: localStorage.getItem('lang') || defaultLang
+      })
+    }
+  }
+  const container = appContainer || cozyUiContainer
+  const { locale } = JSON.parse(container.dataset.cozy)
+
+  return /^\{\{\..*\}\}$/.test(locale) ? defaultLang : locale
```

Comment:
nit: Hard to understand

Reply:
c'est un copier/coller de l'existant, peut-être faire un refacto plus tard pour mutualiser 🤔